R/two.step.hybrid.R

```diff
@@ -1,170 +1,412 @@
-        this.features<-batchwise[[batch.i]]$features2
-        this.medians<-apply(this.fake[,-1:-4],1,median)
-        
-        orig.time<-this.fake[,2]
-        adjusted.time<-fake2[[batch.i]][,2]
-        
-        this.pk.time<-this.aligned<-matrix(0, nrow=nrow(fake3$aligned.ftrs), ncol=ncol(this.fake)-4)
-        
-        # adjusting the time (already within batch adjusted)
-        
-        for(j in 1:length(this.features))
-        {
-            for(i in 1:nrow(this.features[[j]]))
-            {
-                diff.time<- abs(orig.time - this.features[[j]][i,2])
-                sel<-which(diff.time == min(diff.time))[1]
-                this.features[[j]][i,2] <- adjusted.time[sel]
-            }
+  }
+  return(within_batch$corrected_features)
+}
+
+compute_intensity_medians <- function(feature_table) {
+  stopifnot("sample_intensity" %in% colnames(feature_table))
+  feature_table <- dplyr::group_by(feature_table, mz, rt) %>%
+    dplyr::mutate(intensity = median(sample_intensity)) %>%
+    dplyr::ungroup()
+  return(feature_table)
+}
+
+bind_batch_label_column <- function(filenames, metadata) {
+  stopifnot(nrow(metadata) == length(filenames))
+
+  filenames <- as_tibble(filenames)
+  colnames(filenames)[1] <- "filename"
+  filenames <- mutate(filenames, sample_name = get_sample_name(filename))
+  filenames <- inner_join(filenames, metadata, by = "sample_name")
+  return(dplyr::select(filenames, -sample_name))
+}
+
+feature_recovery <- function(cluster,
+                             step_one_features,
+                             batchwise,
+                             filenames_batchwise,
+                             corrected,
+                             aligned,
+                             batches_idx,
+                             mz.tol,
+                             batch.align.mz.tol,
+                             batch.align.chr.tol,
+                             recover.mz.range,
+                             recover.chr.range,
+                             use.observed.range,
+                             min.bw,
+                             max.bw,
+                             recover.min.count) {
+  recovered <- tibble(
+    mz = numeric(),
+    rt = numeric(),
+    mz_min = numeric(),
+    mz_max = numeric()
+  )
+
+  for (batch_id in batches_idx)
+  {
+    this.fake <- step_one_features[[batch_id]]
+    this.fake.medians <- distinct(this.fake, mz, rt, intensity)$intensity
+    this.fake <- long_to_wide_feature_table(this.fake)
+
+    # adjusting the time (already within batch adjusted)
+    this.features <- readjust_times(batchwise[[batch_id]], corrected[[batch_id]])
+
+    aligned_intensities <- dplyr::select(aligned, contains("_intensity"))
+    batchwise_intensities <- dplyr::select(this.fake, contains("_intensity"))
+
+    this.fake.time <- dplyr::select(this.fake, contains("_rt"))
+    this.pk.time <- this.aligned <- matrix(0, nrow = nrow(aligned), ncol = ncol(batchwise_intensities))
```

Comment:
What does `pk` stand for here?

Reply:
No idea. This is the original variable name.